{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "4ac0f68f-d04e-4a26-859b-d58aebcd1a3a",
   "metadata": {},
   "source": [
    "This script migrates the experiments after the BEAST2 package detection has been changed to also look at the \"required\" field."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "a9ba314d-6a33-4819-afd3-5c6ebf8e4969",
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "from io import BytesIO\n",
    "\n",
    "from tqdm import tqdm\n",
    "\n",
    "import phylodata\n",
    "from phylodata.data_types import FileType\n",
    "from phylodata.maintenance.create_new_version import create_new_version\n",
    "\n",
    "from phylodata.process.evolutionary_model.parse_evolutionary_model import (\n",
    "    parse_evolutionary_model,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "fc183902-3df4-4203-8e95-844b620c06d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "EXPERIMENT_IDS = [\n",
    "    \"barrett-2024-mosaic-epo4\",\n",
    "    \"wilinski-2024-congruity-6pcr\",\n",
    "    \"wilinski-2024-congruity-fcth\",\n",
    "    \"imfeld-2024-diversification-xtpf\",\n",
    "    \"imfeld-2024-diversification-slkl\",\n",
    "    \"imfeld-2024-diversification-wu1h\",\n",
    "    \"imfeld-2024-diversification-i5i3\",\n",
    "    \"magalhaes-2024-complete-njnd\",\n",
    "    \"wutke-2024-phylogenomics-5qh4\",\n",
    "    \"wutke-2024-phylogenomics-6eyi\",\n",
    "    \"wutke-2024-phylogenomics-1p3a\",\n",
    "    \"wutke-2024-phylogenomics-jeno\",\n",
    "    \"wutke-2024-phylogenomics-oqvj\",\n",
    "    \"wutke-2024-phylogenomics-ck2k\",\n",
    "    \"wutke-2024-phylogenomics-b462\",\n",
    "    \"wutke-2024-phylogenomics-a4w7\",\n",
    "    \"wutke-2024-phylogenomics-0wb6\",\n",
    "    \"wutke-2024-phylogenomics-05x2\",\n",
    "    \"vieira-2024-mitochondrial-cg9a\",\n",
    "    \"serrano-2024-targeted-zc3e\",\n",
    "    \"serrano-2024-targeted-yd14\",\n",
    "    \"titus-2024-topology-qk0s\",\n",
    "    \"miri-2024-first-g6o6\",\n",
    "    \"rozo-2024-polarization-9ofb\",\n",
    "    \"harrington-2024-dispersal-xw9b\",\n",
    "    \"harrington-2024-dispersal-wn6u\",\n",
    "    \"harrington-2024-dispersal-we1f\",\n",
    "    \"tanoyo-2024-systematics-frn3\",\n",
    "    \"tanoyo-2024-systematics-5flw\",\n",
    "    \"tanoyo-2024-systematics-cskv\",\n",
    "    \"tanoyo-2024-systematics-3ytw\",\n",
    "    \"mcguire-2023-species-cddh\",\n",
    "    \"mcguire-2023-species-iyhy\",\n",
    "    \"mcguire-2023-species-y2km\",\n",
    "    \"pyrcz-2023-phylogeny-53vj\",\n",
    "    \"letsch-2023-jumping-rg34\",\n",
    "    \"thomas-2023-multiple-elyn\",\n",
    "    \"auderset-2023-subgrouping-nq9a\",\n",
    "    \"auderset-2023-subgrouping-uj71\",\n",
    "    \"auderset-2023-subgrouping-eb04\",\n",
    "    \"auderset-2023-subgrouping-l8go\",\n",
    "    \"rez-2023-species-ejki\",\n",
    "    \"rez-2023-species-3b6y\",\n",
    "    \"meyer-2023-morrison-zdih\",\n",
    "    \"mathers-2023-hybridisation-vwt0\",\n",
    "    \"leach-2023-repeated-zn4e\",\n",
    "    \"puentes-2023-pre-y0hv\",\n",
    "    \"forni-2022-macroevolutionary-r34r\",\n",
    "    \"bler-2022-phylogenetic-k54f\",\n",
    "    \"davis-2022-population-f3j9\",\n",
    "    \"kim-2022-phylogenomics-w5kg\",\n",
    "    \"buckingham-2022-population-1mid\",\n",
    "    \"buckingham-2022-population-wksf\",\n",
    "    \"magalhaes-2022-phylogeny-iwk9\",\n",
    "    \"obiol-2022-palaeoceanographic-7e7s\",\n",
    "    \"obiol-2022-palaeoceanographic-cf5c\",\n",
    "    \"obiol-2022-palaeoceanographic-l2ba\",\n",
    "    \"obiol-2022-palaeoceanographic-fpjv\",\n",
    "    \"obiol-2022-palaeoceanographic-u1tx\",\n",
    "    \"obiol-2022-palaeoceanographic-rmgd\",\n",
    "    \"brownstein-2022-hidden-zhs4\",\n",
    "    \"brownstein-2022-hidden-nk9h\",\n",
    "    \"johnston-2022-cophylogeny-2u2d\",\n",
    "    \"darlim-2022-impact-umzp\",\n",
    "    \"darlim-2022-impact-em28\",\n",
    "    \"sancho-2022-tracking-8zg6\",\n",
    "    \"takahashi-2022-stable-rt26\",\n",
    "    \"gable-2022-genomic-n5f4\",\n",
    "    \"gable-2022-genomic-nugh\",\n",
    "    \"gable-2022-genomic-fm05\",\n",
    "    \"anzas-2022-bayesian-p68i\",\n",
    "    \"anest-2021-evolving-n92s\",\n",
    "    \"johnson-2021-systematics-dbeu\",\n",
    "    \"johnson-2021-systematics-3ch0\",\n",
    "    \"johnson-2021-systematics-ex86\",\n",
    "    \"brock-2021-color-a2u8\",\n",
    "    \"leach-2021-phylogenomic-aphe\",\n",
    "    \"leach-2021-phylogenomic-ahe1\",\n",
    "    \"mil-2021-new-786l\",\n",
    "    \"mil-2021-new-j2s4\",\n",
    "    \"pela-2021-subterranean-zego\",\n",
    "    \"pela-2021-subterranean-iao8\",\n",
    "    \"blow-2021-molecular-l62k\",\n",
    "    \"near-2021-phylogeny-dfe7\",\n",
    "    \"near-2021-phylogeny-hiv3\",\n",
    "    \"arbour-2021-little-vfxn\",\n",
    "    \"arbour-2021-little-33ff\",\n",
    "    \"catanach-2021-systematics-6ujn\",\n",
    "    \"catanach-2021-systematics-yiig\",\n",
    "    \"liu-2021-testing-me58\",\n",
    "    \"hara-2021-relict-bi5r\",\n",
    "    \"sidlauskas-2021-total-gcfr\",\n",
    "    \"savelyev-2020-bayesian-88zc\",\n",
    "    \"vianna-2020-genome-okrw\",\n",
    "    \"vianna-2020-genome-mze8\",\n",
    "    \"corrales-2020-allopatric-v6pl\",\n",
    "    \"corrales-2020-allopatric-w7lu\",\n",
    "    \"stervander-2020-molecular-0h8r\",\n",
    "    \"chaves-2020-evolutionary-w9n2\",\n",
    "    \"chaves-2020-evolutionary-vubq\",\n",
    "    \"chaves-2020-evolutionary-5dgl\",\n",
    "    \"chaves-2020-evolutionary-8byo\",\n",
    "    \"chaves-2020-evolutionary-odm3\",\n",
    "    \"chaves-2020-evolutionary-9cs6\",\n",
    "    \"chaves-2020-evolutionary-jjat\",\n",
    "    \"chaves-2020-evolutionary-fmyh\",\n",
    "    \"chaves-2020-evolutionary-slrw\",\n",
    "    \"chaves-2020-evolutionary-0vrl\",\n",
    "    \"deli-2020-subterranean-9yvn\",\n",
    "    \"buckley-2020-multiple-mpex\",\n",
    "    \"burridge-2020-migration-nbmn\",\n",
    "    \"tsang-2020-dispersal-9tzw\",\n",
    "    \"oliver-2020-oligocene-cv1m\",\n",
    "    \"linck-2020-speciation-ncxu\",\n",
    "    \"imfeld-2020-mitochondrial-stuu\",\n",
    "    \"kornilios-2019-genome-3ijw\",\n",
    "    \"slater-2019-hierarchy-xv5v\",\n",
    "    \"hancock-2019-phylogeography-4z1v\",\n",
    "    \"brandrud-2019-phylogenomic-73ad\",\n",
    "    \"brandrud-2019-phylogenomic-iyvq\",\n",
    "    \"brandrud-2019-phylogenomic-q6h8\",\n",
    "    \"presslee-2019-palaeoproteomics-btvn\",\n",
    "    \"presslee-2019-palaeoproteomics-mjp2\",\n",
    "    \"munro-2019-climate-6tvf\",\n",
    "    \"piatkowski-2019-functional-7jsc\",\n",
    "    \"nen-2019-postglacial-qh0e\",\n",
    "    \"nen-2019-postglacial-n1bf\",\n",
    "    \"economo-2019-evolution-vn9j\",\n",
    "    \"leach-2019-exploring-rb8o\",\n",
    "    \"leach-2019-exploring-1px9\",\n",
    "    \"stervander-2019-origin-1z13\",\n",
    "    \"stervander-2019-origin-avfu\",\n",
    "    \"stervander-2019-origin-702n\",\n",
    "    \"stange-2018-bayesian-q044\",\n",
    "    \"stange-2018-bayesian-krv3\",\n",
    "    \"stange-2018-bayesian-2uhx\",\n",
    "    \"clucas-2018-comparative-sqdr\",\n",
    "    \"clucas-2018-comparative-ic87\",\n",
    "    \"toljagic-2018-millions-wb2s\",\n",
    "    \"jorge-2018-getting-7s4n\",\n",
    "    \"j-2018-allegory-iwn6\",\n",
    "    \"marburger-2018-whole-8zqj\",\n",
    "    \"zaher-2018-origin-11vu\",\n",
    "    \"blanchet-2018-related-q4yw\",\n",
    "    \"salariato-2018-reinstatement-b34a\",\n",
    "    \"singhal-2018-population-v2of\",\n",
    "    \"carnicero-2018-phylogeography-zpjw\",\n",
    "    \"carnicero-2018-phylogeography-riax\",\n",
    "    \"varela-2018-phylogeny-uvh2\",\n",
    "    \"trotta-2018-community-a38u\",\n",
    "    \"grear-2018-inferring-p7zg\",\n",
    "    \"grear-2018-inferring-vv5k\",\n",
    "    \"grear-2018-inferring-ltdj\",\n",
    "    \"grear-2018-inferring-86ph\",\n",
    "    \"grear-2018-inferring-plz2\",\n",
    "    \"lajeunesse-2018-systematic-307m\",\n",
    "    \"sessa-2018-evolution-l9k2\",\n",
    "    \"sessa-2018-evolution-8p92\",\n",
    "    \"shu-2017-multi-7trv\",\n",
    "    \"sklen-2017-phylogeny-qwwe\",\n",
    "    \"sklen-2017-phylogeny-nda2\",\n",
    "    \"sklen-2017-phylogeny-l1mi\",\n",
    "    \"salariato-2017-climatic-4as2\",\n",
    "    \"salariato-2017-climatic-5lpn\",\n",
    "    \"slater-2017-independent-xt7u\",\n",
    "    \"saladin-2017-fossils-lfr1\",\n",
    "    \"saladin-2017-fossils-008y\",\n",
    "    \"saladin-2017-fossils-wtv8\",\n",
    "    \"saladin-2017-fossils-s3c2\",\n",
    "    \"saladin-2017-fossils-q2ha\",\n",
    "    \"saladin-2017-fossils-ju2l\",\n",
    "    \"saladin-2017-fossils-kgu9\",\n",
    "    \"saladin-2017-fossils-jfg5\",\n",
    "    \"saladin-2017-fossils-yv8r\",\n",
    "    \"saladin-2017-fossils-ao96\",\n",
    "    \"bagley-2016-phylogeography-18cm\",\n",
    "    \"bagley-2016-phylogeography-tsuj\",\n",
    "    \"close-2016-mosaicism-kq3y\",\n",
    "    \"close-2016-mosaicism-6hm9\",\n",
    "    \"close-2016-mosaicism-d9zx\",\n",
    "    \"close-2016-mosaicism-zkjh\",\n",
    "    \"rawlence-2016-human-852c\",\n",
    "    \"rawlence-2016-human-ojb6\",\n",
    "    \"salariato-2016-diversification-hhr5\",\n",
    "    \"salariato-2016-diversification-6296\",\n",
    "    \"salariato-2016-diversification-059x\",\n",
    "    \"cole-2016-nearctic-fgy6\",\n",
    "    \"foote-2016-genome-10ig\",\n",
    "    \"foote-2016-genome-hcha\",\n",
    "    \"foote-2016-genome-8xm0\",\n",
    "    \"foote-2016-genome-ydzw\",\n",
    "    \"foote-2016-genome-1k2i\",\n",
    "    \"v-2016-new-g4qu\",\n",
    "    \"looney-2016-tropics-5f11\",\n",
    "    \"zarza-2016-hidden-0uci\",\n",
    "    \"tornabene-2016-repeated-k72b\",\n",
    "    \"bapst-2016-topology-7lg1\",\n",
    "    \"bapst-2016-topology-d1nj\",\n",
    "    \"vuataz-2016-molecular-9dq5\",\n",
    "    \"jr-2015-multilocus-qbd4\",\n",
    "    \"jr-2015-multilocus-t1nb\",\n",
    "    \"winger-2015-inferring-g4kz\",\n",
    "    \"paun-2015-processes-sqks\",\n",
    "    \"close-2015-evidence-mmit\",\n",
    "    \"morin-2015-geographic-orf9\",\n",
    "    \"colombo-2015-diversity-4yr1\",\n",
    "    \"colombo-2015-diversity-5jap\",\n",
    "    \"colombo-2015-diversity-r5ri\",\n",
    "    \"colombo-2015-diversity-cf8w\",\n",
    "    \"colombo-2015-diversity-i28o\",\n",
    "    \"colombo-2015-diversity-gwrh\",\n",
    "    \"colombo-2015-diversity-inb1\",\n",
    "    \"colombo-2015-diversity-xdg9\",\n",
    "    \"colombo-2015-diversity-8kcc\",\n",
    "    \"colombo-2015-diversity-ne95\",\n",
    "    \"colombo-2015-diversity-4xoe\",\n",
    "    \"colombo-2015-diversity-xvrj\",\n",
    "    \"hodges-2014-phylogeography-573d\",\n",
    "    \"gohli-2014-evolutionary-gtlx\",\n",
    "    \"gohli-2014-evolutionary-scpc\",\n",
    "    \"hammer-2014-multigene-sfjz\",\n",
    "    \"georges-2013-contemporary-sr8q\",\n",
    "]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7ffdc0e4-1583-4a20-a0c7-10effec37c94",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 37%|█████████████████████████████████████████▏                                                                    | 83/222 [00:12<00:06, 20.01it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "near-2021-phylogeny-dfe7\n",
      "[EvolutionaryModelComponent(name='SA', type=<ModelType.TREE_PRIOR: 'treePrior'>, parameters={})]\n",
      "(old: [] )\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 41%|█████████████████████████████████████████████▌                                                                | 92/222 [00:27<01:11,  1.82it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "savelyev-2020-bayesian-88zc\n",
      "[EvolutionaryModelComponent(name='SA', type=<ModelType.TREE_PRIOR: 'treePrior'>, parameters={})]\n",
      "(old: [] )\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 52%|████████████████████████████████████████████████████████▉                                                    | 116/222 [00:39<00:20,  5.14it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "slater-2019-hierarchy-xv5v\n",
      "[EvolutionaryModelComponent(name='SA', type=<ModelType.TREE_PRIOR: 'treePrior'>, parameters={})]\n",
      "(old: [] )\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 52%|████████████████████████████████████████████████████████▉                                                    | 116/222 [00:54<00:20,  5.14it/s]"
     ]
    }
   ],
   "source": [
    "for experiment_id in tqdm(EXPERIMENT_IDS):\n",
    "    experiment = phylodata.load_experiment(\n",
    "        experiment_id, files_to_download=[FileType.BEAST2_CONFIGURATION], directory=\"../data\"\n",
    "    )\n",
    "\n",
    "    xml_file = experiment.get_file_of_type(FileType.BEAST2_CONFIGURATION)\n",
    "    xml_bytes = BytesIO(xml_file.local_path.read_bytes())\n",
    "\n",
    "    models = parse_evolutionary_model(xml_bytes)\n",
    "\n",
    "    if len(models) != len(experiment.evolutionary_model):\n",
    "        print(experiment_id)\n",
    "        print(models)\n",
    "        print(\"(old:\", experiment.evolutionary_model, \")\")\n",
    "\n",
    "        experiment.evolutionary_model = models\n",
    "        create_new_version(experiment)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
